src/main/java/core/basesyntax/WorkWithFile.java

```diff
@@ -1,8 +1,62 @@
 package core.basesyntax;
 
+import java.io.BufferedReader;
+import java.io.BufferedWriter;
+import java.io.FileReader;
+import java.io.FileWriter;
+import java.io.IOException;
+
 public class WorkWithFile {
+    private static final int OPERATION_TYPE_INDEX = 0;
+    private static final int AMMOUNT_INDEX = 1;
+    private static final int BUY_INDEX = 0;
+    private static final int SUPPLY_INDEX = 1;
+    private static final String SEPARATOR = ",";
 
     public void getStatistic(String fromFileName, String toFileName) {
+        String[] value = readFromFile(fromFileName).split(SEPARATOR);
+
+        String builder = "supply," + value[SUPPLY_INDEX] + System.lineSeparator()
+                + "buy," + value[BUY_INDEX] + System.lineSeparator()
+                + "result,"
+                + (Integer.parseInt(value[SUPPLY_INDEX]) - Integer.parseInt(value[BUY_INDEX]));
+
+        writeToFile(toFileName, builder);
+    }
+
+    private String readFromFile(String fromFileName) {
+        int buy = 0;
+        int supply = 0;
+
+        try (BufferedReader bufferedReader = new BufferedReader(new FileReader(fromFileName))) {
+            String value = bufferedReader.readLine();
+
+            while (value != null) {
+                String[] lineValue = value.split(SEPARATOR);
+                switch (lineValue[OPERATION_TYPE_INDEX]) {
+                    case "buy":
+                        buy += Integer.parseInt(lineValue[AMMOUNT_INDEX]);
+                        break;
+                    case "supply":
+                        supply += Integer.parseInt(lineValue[AMMOUNT_INDEX]);
+                        break;
+                    default:
+                        break;
+                }
+                value = bufferedReader.readLine();
+            }
```

Comment:
Same about this method. If It has a name like `readFromFile` it must only read from file, and return the String value.
All other logic relates to the report generation

Reply:
+